packages/react-badge/src/components/Badge/Badge.types.ts

```diff
@@ -1,11 +1,11 @@
 import * as React from 'react';
-import type { ComponentProps, ComponentState, ObjectShorthandProps } from '@fluentui/react-utilities';
+import type { ComponentProps, ComponentSlotProps, ComponentState } from '@fluentui/react-utilities';
 
 export type BadgeSlots = {
   // react has a non-standard `color` attribute in its types
   // https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a4ab0fa432320e70da9e51c8ae2e47377f65804b/types/react/index.d.ts#L1868
-  root: ObjectShorthandProps<Omit<React.HTMLAttributes<HTMLElement>, 'color'>>;
-  icon?: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>>;
+  root: ComponentSlotProps<React.ComponentType<Omit<React.HTMLAttributes<HTMLElement>, 'color'>>>;
+  icon?: ComponentSlotProps<React.ComponentType<React.HTMLAttributes<HTMLElement>>>;
```

Comment:
It looks like Badge needs a bit of cleanup in general... But these should use `IntrinsicShorthandProps`:
```suggestion
  root: Omit<IntrinsicShorthandProps<'div'>, 'color'>;
  icon?: IntrinsicShorthandProps<'span'>;
```